Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -410,3 +469,14 @@ extension LoginDetailViewController: LoginTableViewCellDelegate {
         return false
     }
 }
+
+extension LoginDetailViewController: PasscodeEntryDelegate {
+    func passcodeValidationDidSucceed() {
+        self.removeBackgroundedBlur()
+        self.navigationController?.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    func userDidCancelValidation() {
+        self.navigationController?.popToRootViewControllerAnimated(false)
+    }
+}
```

Comment:
I wanted to try avoiding a subclass approach in case we want to use this functionality for another type of view controller (UICollectionViewController, UITableViewController). Instead, I've managed to extract all the logic into a protocol extension with constraints on UIViewController! See the follow up commit.